### PR TITLE
Armor Layers Additions and Improvements

### PR DESCRIPTION
**Pull request by DavidKeaton**
More UI improvements! This time for the Armor Layers [+] screen! 

Features:
- Added ability to equip or unequip items from the screen itself. It will unequip the item based on the location of the cursor, and equip an item above the item selected when equipping! Also, when equipping an item, no currently worn items are shown!
- Adjusted clothing properties display. There are 2 headers now, `Properties` and `Protection`, and `Protection` now shows `Environmental` protection!
- Assigning invlets to armor is now with a prompt (yay?), as I rarely use invlets unless _I_ assign them, and having to clear them all makes me go "AGGHHHH!" (of course, I hope to speak for others like me, not just me).
- Prettified the [?] screen for Armor Layers, making it a bit easier to tell key from description.
- Now taking off/putting on clothing takes time, with an `activity_handler` no less!

### CODE REVIEW

File: src/armor_layers.cpp [hunk 1]
```diff
@@ -305,29 +308,65 @@ void player::sort_armor()
...
+        } else if (action == "EQUIP_ARMOR") {
+            // filter inventory for all items that are armor/clothing
+            int pos = g->inv_for_unequipped(_("Put on:"), [](const item &it) {
+                        return it.is_armor();
+                    });
+            // only equip if something valid selected!
+            if(pos != INT_MIN) {
+                // wear the item, if error, show it
+                if(!wear(pos)) {
+                    debugmsg("Failed trying to equip item @index [%d]", pos);
```
Comment (BevapDin) [inline src/armor_layers.cpp]: This is actually *very* likely, try wearing something while already wearing power armor. Or try wearing a second pair of shoes etc.

Comment (DavidKeaton) [inline src/armor_layers.cpp]: Doy, I even browsed through all that code and saw the power armour return false. my b.

**Comment by Zireael07** [comment]
Wooot!

**Comment by DavidKeaton** [comment]
pic2
pic1

**Comment by BevapDin** [comment]
`player::wear_item` reduces the players movement points by 350 (`player::takeoff` doesn't, there it'S handled in `game::takeoff`), this might leave the player with a lot negative movement points and no actions for several turns. Have you considered this?

**Comment by DavidKeaton** [comment]
this I had not considered, thanks for the heads up!

**Comment by KA101** [comment]
This will be well-received, once the time issue is worked out.  Thinking long-action protection ought to suffice, but IIRC implementing that for AIM wasn't trivial and wouldn't expect it to be here either.  :-/

**Comment by DavidKeaton** [comment]
I'll figure something out. :-)

**Comment by DavidKeaton** [comment]
A zombie approached me as I was ripping off clothing, this look any better? :-P

**Comment by kevingranade** [comment]
What I did with the aiming menu was set an action that has the effect of
re-entering the menu, then exit the menu, this runs all the safe mode code
and allows interruptions if you run out of move points.

**Comment by DavidKeaton** [comment]
@kevingranade okay that's a very intriguing way of doing that, and sounds so straightforward. I think I essentially captured (see: stole) the same idea.

**Comment by DavidKeaton** [comment]
should be ready2merge whenevs :-)

**Comment by KA101** [comment]
I'll be heading out of town for most of today but back this afternoon/evening.

**Comment by DavidKeaton** [comment]
Have a safe trip @KA101 !

**Comment by KA101** [comment]
Meh, lecturing on stuff I already knew, major argument that doesn't feel like it accomplished anything productive, and I see Kevin had to merge everything.  Waste of a day.

Time to merge.

File: src/character.h [hunk 2]
```diff
@@ -91,6 +91,17 @@ class Character : public Creature
         {
             return -2 - position;
         }
+
+        // checks to see if an item is worn
+        bool is_worn(const item &thing) const
+        {
+            for(const auto &elem : worn) {
```
Comment (bkentel) [inline src/character.h]: How about `std::any_of`? Also, why place this in the header?

**Comment by Snaaty** [comment]
This is great work.

File: src/character.h (hunk 2 above)
Comment (DavidKeaton) [inline src/character.h]: Most of C++'s std library is a learn by use basis for me. I come from C and Assembler primarily, this is really the first time using C++ in a project. :-)

**Comment by DavidKeaton** [comment]
@Snaaty thanks man :-)

**Pull request merged**